cnd/src/facade.rs

```diff
@@ -137,14 +146,46 @@ impl<B: ledger::bitcoin::Bitcoin + 'static> HtlcEvents<B, asset::Bitcoin> for Fa
 }
 
 #[async_trait::async_trait]
-impl<A> HtlcEvents<Ethereum, A> for Facade
-where
-    A: Asset + Send + Sync + 'static,
-    ethereum::Cache<Web3Connector>: HtlcEvents<Ethereum, A>,
-{
+impl HtlcEvents<Ethereum, asset::Ether> for Facade {
```

Comment:
Would it be possible to avoid this kind of refactoring as part of this PR?

Reply:
Unfortunately not, because I hit a compiler error that forbids to abstract over traits the crate doesn't own.

Reply:
Indeed, I should have spent more time trying to understand.